tsdb/index/tsi1/index.go

```diff
@@ -845,7 +946,19 @@ func (i *Index) TagValueSeriesIDIterator(name, key, value []byte) (tsdb.SeriesID
 			a = append(a, itr)
 		}
 	}
-	return tsdb.MergeSeriesIDIterators(a...), nil
+
+	itr := tsdb.MergeSeriesIDIterators(a...)
+	if !EnableBitsetCache {
+		return itr, nil
+	}
+
+	// Check if the iterator contains only series id sets. Cache them...
+	if ssitr, ok := itr.(tsdb.SeriesIDSetIterator); ok {
+		ss := ssitr.SeriesIDSet()
+		ss.SetCOW(true) // This is important to speed the clone up.
+		i.tagValueCache.Put(name, key, value, ss)
```

Comment:
seems like concurrent calls to `TagValueSeriesIDIterator` can cause that debugging condition in `Put` to happen. Maybe either do the `RLock` + check + `Lock` + check + insert song and dance, or maybe it's ok to just drop it on the floor in the `Put` call since it should be the same value anyway?

Reply:
We have to `Lock` the cache anyway because we have to put the item at the beginning of the eviction queue. So even if we drop the item on the floor we still would have had to have `Lock`ed the cache.